test/e2e/flask/multichain-api/create-session.spec.ts

```diff
@@ -46,4 +61,144 @@ describe('Multichain API', function () {
       );
     });
   });
+
+  describe('Connect wallet to the multichain dapp via `externally_connectable`, call `wallet_createSession` with requested EVM scope that match the user’s enabled networks', function () {
+    it('should only select the specified EVM scopes requested by the user, and session scope contain valid accounts only', async function () {
+      await withFixtures(
+        {
+          title: this.test?.fullTitle(),
+          fixtures: new FixtureBuilder().withPopularNetworks().build(),
+          ...DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+        },
+        async ({
+          driver,
+          extensionId,
+        }: {
+          driver: Driver;
+          extensionId: string;
+        }) => {
+          const INVALID_ACCOUNT = '0x9999999999999999999999999999999999999999';
+          const requestScopesToNetworkMap = {
+            'eip155:1': 'Ethereum Mainnet',
+            'eip155:42161': 'Arbitrum One',
+            'eip155:10': 'OP Mainnet',
+          };
+
+          const requestScopes = Object.keys(requestScopesToNetworkMap);
+          const networksToRequest = Object.values(requestScopesToNetworkMap);
+
+          await openMultichainDappAndConnectWalletWithExternallyConnectable(
+            driver,
+            extensionId,
+          );
+
+          await addRequestAccountsToCreateSession(driver, [
+            DEFAULT_FIXTURE_ACCOUNT,
```

Comment:
In this test we should use an account that isn't the first one in the wallet since this one will be selected by default whether or not it is requested

Reply:
Let me know if this doesn't make sense or you don't know what I'm talking about haha

Reply:
resolved in [de15bfb](https://github.com/MetaMask/metamask-extension/pull/29356/commits/de15bfb58aa908ad3b7f54092a123b53e963a270)